src/views/AddItem.jsx

```diff
@@ -12,17 +12,67 @@ export function AddItem({ listToken }) {
 		addItemToList();
 	};
 
+	// Removes excess spaces or special characters, and makes all leters lowercase.
+	const normalizeText = (item) => {
+		return item
+			.trim() // removes space before and after item
+			.split(/ +/) // removes extra spaces between words
+			.join(' ')
+			.toLowerCase() // makes text all lowercase
+			.replace(/[^a-z0-9]/gi, ''); // remove punctuation and special characters
+	};
+
+	// Checks if user's input item already exists in the database.
+	const checkIfItemExists = (item) => {
+		for (let i = 0; i < data.length; i++) {
+			// If the current database item (normalized) is the same as the user's input item (normalized)
+			if (normalizeText(data[i].name) === normalizeText(itemName)) {
+				return true;
+			}
+		}
+		return false;
+	};
+
 	const addItemToList = async () => {
-		const trimmedItemName = itemName.trim();
+		const normalizedItemName = normalizeText(itemName);
+
 		try {
-			await addItem(listToken, {
-				itemName: trimmedItemName,
-				daysUntilNextPurchase,
-			});
-			setSubmitStatus({
-				type: 'success',
-				value: 'Item was successfully saved to the database',
-			});
+			// Check if the item is already on the user's list.
+			if (checkIfItemExists(normalizedItemName)) {
+				setSubmitStatus({
+					type: 'error',
+					value: 'You already added this item to your list.',
+				});
+				setTimeout(() => {
+					setSubmitStatus({
+						type: 'idle',
+						value: '',
+					});
+				}, 3000);
+
+				// Check if the user is trying to submit an empty item.
+			} else if (normalizedItemName === '') {
+				setSubmitStatus({
+					type: 'error',
+					value: 'Please enter an item.',
+				});
+				setTimeout(() => {
+					setSubmitStatus({
+						type: 'idle',
+						value: '',
+					});
```

Comment:
We're reusing this twice, maybe an opportunity to break this out into another function that we can call twice?

Reply:
+1 to Olivia's idea. Nice to keep things DRY and, more importantly imo, ETC aka easy to change. If you have reusable function and decided you wanted to change the length of the timeout it becomes much easier.

Reply:
Yay thank you, we ❤️ DRY code!